### PR TITLE
cmd, common, crypto, keystore: unlock keys by command line

### LINKED ISSUE #421: keystore: unlock keys and save keypair in memory
need to add functionality to unlock keys in the disk keystore. we should provide a command line flag `--unlock` to prompt a the user to unlock an account. eg:
```
gossamer account --unlock=[address]
> type password to unlock account: ***
```

could also use the password flag to do this without a prompt:
` gossamer account --unlock=[address] --password=mypassword`

after the account has been unlocked, the public/private keypair should be stored in memory in the keystore.
also need to make sure the private keys get zero'd out after they are done being used.

### PR DESCRIPTION
**Pull request by noot**
## Changes
- add `--unlock` flag, upon starting gossamer can use `gossamer --unlock=0,1 --password=pwd0,pwd1` to unlock keys and put them in the keystore
- if no passwords are specified with the `--password` flag, the user is prompted to enter their password

## Tests:
```
go test ./cmd/...
```

```
cd cmd/gossamer/ && go install && cd ../..
gossamer account --generate
gossamer --unlock=0
gossamer --unlock=0 --password=[mypwd]
```

### Issues:
closes #421

### CODE REVIEW

File: cmd/gossamer/account.go [hunk 1]
```diff
@@ -49,12 +49,14 @@ func handleAccounts(ctx *cli.Context) error {
...
 		if flagtype := ctx.Bool(utils.Sr25519Flag.Name); flagtype {
-			keytype = utils.Sr25519KeyType
+			keytype = crypto.Sr25519Type
 		} else if flagtype := ctx.Bool(utils.Ed25519Flag.Name); flagtype {
-			keytype = utils.Ed25519KeyType
-		}
+			keytype = crypto.Ed25519Type
+		} /*else if flagtype := ctx.Bool(utils.Secp256k1Flag.Name); flagtype {
+			keytype = crypto.Secp256k1Type
+		}*/
```
Comment (ryanchristo) [inline cmd/gossamer/account.go]: Are we waiting to add secp?

File: cmd/gossamer/account.go [hunk 2]
```diff
@@ -127,11 +129,27 @@ func importKey(filename, datadir string) (string, error) {
...
+}
+
+// getKeyFiles returns the filenames of all the keys in the datadir's keystore
+func getKeyFiles(datadir string) ([]string, error) {
 	keystorepath, err := keystoreDir(datadir)
 	if err != nil {
 		return nil, fmt.Errorf("could not get keystore directory: %s", err)
 	}
 
+	fmt.Printf("getKeyFiles datadir %s\n", keystorepath)
```
Comment (ryanchristo) [inline cmd/gossamer/account.go]: Should this be a log?

File: cmd/gossamer/account.go [hunk 3]
```diff
@@ -178,6 +195,10 @@ func generateKeypair(keytype, datadir string, password []byte) (string, error) {
 		}
 	}
 
+	// if !common.Exists(keystorepath) {
+	// 	os.Mkdir(keystorepath, os.ModePerm)
+	// }
```
Comment (ryanchristo) [inline cmd/gossamer/account.go]: Should this be removed or updated?

File: cmd/gossamer/account.go (hunk 1 above)
Comment (noot) [reply cmd/gossamer/account.go]: yeah, I can uncomment this when my other PR gets merged

File: cmd/gossamer/account.go (hunk 2 above)
Comment (noot) [reply cmd/gossamer/account.go]: woops meant to remove

File: cmd/gossamer/account.go (hunk 3 above)
Comment (noot) [reply cmd/gossamer/account.go]: yeah forgot to remove it

File: cmd/gossamer/account.go (hunk 1 above)
Comment (ryanchristo) [reply cmd/gossamer/account.go]: PR merged

**Comment by noot** [comment]
@ryanchristo sorry i messed up when merging :( should be fixed now!

**Comment by noot** [comment]
@ryanchristo nice! I've added the secp256k1 flag and key gen in

**Pull request merged**